src/Symfony/Component/Runtime/RuntimeInterface.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Runtime;
+
+/**
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+interface RuntimeInterface
+{
+    /**
+     * Returns a closure that builds an object holding the userland app and returns it.
```

Comment:
Since naming thing here could be confusing and there are way too many closures for a normal person like me to understand, I suggest we add more descriptions here and maybe even an example.

Same with `getMain()`

Reply:
I was working exactly on this aspect as you wrote the comment :)
I also just pushed some update. Can you please have a look and give me some suggestions?

Reply:
https://github.com/symfony/symfony/pull/36652/files#r420642327